src/classes/CRLP_Rollup_TDTM.cls

```diff
@@ -62,15 +62,17 @@ public class CRLP_Rollup_TDTM extends TDTM_Runnable {
 
         List<SObject> recordsToRecalculate = new List<SObject>();
         for (Integer i=0; i < ((newlist == null ? oldlist.size() : newlist.size())); i++) {
+            SObject record = (newlist == null ? oldlist[i] : newlist[i]);
             if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
                 // Have any of the "key" fields on the record changed?
-                if (isRollupRecalcNeeded(newlist[i], oldlist[i])) {
-                    recordsToRecalculate.add(newlist[i]);
+                if (isRollupRecalcNeeded(record, oldlist[i])) {
+                    recordsToRecalculate.add(record);
+                    recordsToRecalculate.add(oldlist[i]);
```

Comment:
It looks like the fix is to always include the original record whenever a recalc is required. What are your thoughts about only including the original record when a target object id has been changed?

Reply:
I started going down that route originally, but I realized that it was unnecessary. The way the logic works is to loop through the `recordsToRecalculate` collection to get the Account or PrimaryContact Id's into a Set. Then it just calls the recalculate for that set of Id's. With that in mind, it really doesn't matter whether the AccountId or PrimaryContactId was changed or not. If the Id wasn't changed, the end result is the same -- the Id only appears once in the set anyway. If it was changed, now the Set contains both the Old and New Id's. Building a new method to check for the changed value didn't actually add any value to the process.